spring-integration-smb/src/main/java/org/springframework/integration/smb/filters/SmbSimplePatternFileListFilter.java

```diff
@@ -43,4 +47,19 @@ protected String getFilename(SmbFile file) {
 		return (file != null) ? file.getName() : null;
 	}
 
+	/**
+	 * Indicates whether the file is a directory or not.
+	 * @param file SMB file object
+	 * @return true if it's a directory.
+	 * @see AbstractDirectoryAwareFileListFilter#isDirectory(java.lang.Object)
+	 */
+	@Override
+	protected boolean isDirectory(SmbFile file) {
+		try {
+			return file.isDirectory();
+		}
+		catch (SmbException e) {
+			throw new UncheckedIOException(e);
```

Comment:
That's cool, of course, long waiting runtime exception for IO :smiley: 

But I wonder if there is a new version of SMB with better Java 8 adoption that we can just upgrade and go.

Reply:
Yeah, I am not sure about that, there are other libs out there but I highly doubt they are Java 8 

<https://github.com/hierynomus/smbj> is Java 7 and supports SMB2 and SMB3 vs current jCIFS supports CIFS/SMB. Apache Commons VFS also have CIFS (sandbox).

Let me know your thoughts. I think it will not be too hard to swap out the libs.

Reply:
Well, let's fix this issue and maybe release with what we have so far.
And the choice to move to some more modern library let's do afterwards!